dhcpv4/nclient4/lease_test.go

```diff
@@ -236,15 +236,24 @@ func (sll *testServerLeaseList) runTest(t *testing.T) {
 		sll.lastTestSvrErrLock.RLock()
 		keepgoing := chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
 		sll.lastTestSvrErrLock.RUnlock()
-		if keepgoing {
-			err = clnt.Release(lease)
```

Comment:
What is the benefit of this modification?

Reply:
Since we need to test `keepgoing` twice (once for `Renew` and again for `Relese` since we need stop going if either of them fails), doing it this way avoids all nesting and makes the two cases homogeneous. This code for `Release` hasn't been removed, it has just moved down...

Reply:
I guess my question was: Is there any problem with having it like this to keep it consistent?

```
		if keepgoing {
        		lease, err = clnt.Renew(context.Background(), lease.Offer)
			sll.lastTestSvrErrLock.RLock()
			keepgoing = chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
			sll.lastTestSvrErrLock.RUnlock()
		}

		if keepgoing {
	        	err = clnt.Release(lease)
		        //this sleep is to make sure release is handled by server
        		time.Sleep(time.Second)
	        	sll.lastTestSvrErrLock.RLock()
	        	chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
        		sll.lastTestSvrErrLock.RUnlock()
		}
```

Reply:
Aha, I see now. Yes, I can refactor it to be that way. It might be a bit confusing with `keepgoing` being updated inside the first `if`, but nevertheless it works the same.

Reply:
Fixed in 16f5a6e00347d22bc516d4f9c63485d25de4c852.